tox.ini

```diff
@@ -1,18 +1,28 @@
 [tox]
 envlist =
     clean
-    py{312,311,310,39,38}
-    py38-mindeps
+    py{3.12,3.11,3.10,3.9,3.8}
+    py3.8-mindeps
     cov-combine
     cov-report
     mypy
 minversion = 4.3.5
 
 [testenv]
 package = wheel
-wheel_build_env = build_wheel
+# tox-uv currently has a bug that erases the built wheel environment
+# when 'recreate=true' (as it is true when the 'sdkmain' factor is present).
+# To overcome this, it's necessary to specify a different wheel build environment.
+wheel_build_env =
+    !sdkmain: build_wheel
+    sdkmain: build_wheel_sdkmain
 extras = test
 passenv = GLOBUS_SDK_PATH
+# When testing against SDK main it's necessary to recreate the tox environment
+# so that the SDK main tarball is re-downloaded and installed for each run.
+recreate =
+    !sdkmain: false
+    sdkmain: true
```

Comment:
As discussed, :+1: for this workaround! I wish it weren't necessary, but happily it works.